### ISSUE TITLE
Allow extensions to aria announce

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#114718]
We should have an API which extensions can use to announce things to the users via a screen reader.

**Comment by isidorn** [comment]
```
export namespace aria {
	export function announce(message: string): void;
}
```

Should there be a severity? I think not, since this could be misused?

**Comment by isidorn** [comment]
Feedback from the UX meeting: 
* This should maybe live in the same namespace as notifications and not be a seperate thing. Or should just live with a status bar item, and be a field `announce`
* It feels weird to expose some API which we actually do not want extensions to use. Since idealy screen readers should read things based on focus, and based on live-regions
* Worry that extensions might just call this a lot, and it is a bit hard to define guidance when to use this

Due to the feedback I think we should collect more use cases from extensions before we add anything to the proposed api.
Thus pinging @setaskin for more details with potential usage of this in LiveShare.

Moving to backlog until we figure out the exact scenarios.

**Comment by setaskin** [comment]
Hi, yes do have many cases that might have potential use. In below screen shot narrator doesn't announce about deleted items . ( deleting / adding  a participant, server, terminal etc. is not announced by narrator. 
![image](https://user-images.githubusercontent.com/37163639/106722164-779c8180-65ba-11eb-996f-00e8d1665690.png)

**Comment by setaskin** [comment]
we do have some workaround in some cases to make the narrator announce the deleted / added item.  We use info notification  as in the below gif. but since we don't want to distract user with too many notifications we don't use it in every case. below the narrator announce the deleted item 
![Narrator](https://user-images.githubusercontent.com/37163639/106723010-8172b480-65bb-11eb-9f9a-8f9209c874d4.gif)

**Comment by isidorn** [comment]
@setaskin thanks for providing more details. 
From my experience when talking to our accessibility community all of the users use NVDA on Windows and not Narattor, so for us it would be much more interesting to review issues with NVDA.

I think the pattern of showing notifications quickly just for a short progress and to get something read is not that nice. Since visual users might be distracted, and I believe most of the time this notification can be very short thus only being a distracting flash.

This is very similar to how we handle deleting files in the File Explorer in VS Code. We do not announce that a file got deleted, imho it is pretty obvious that is about to happen since the user triggered that action.
So for the delete I would argue you do not need to announce anything, once the item gets removed from the tree it will be pretty obvious to the user what is going on.

Some things which could be improved from your gif:
* once an item gets removed, the focus should move up, and not on the whole tree. That is unexpected. You can check how this behaves in VS Code explorer

Some other actions in your scenario might need to be announced, and maybe for them we can use silent notifications, and we can potentialy introduce a flag for silent notifications if they should be read out or not. Silent notifications only appear in the status bar.

Let me know what you think, and if my suggestions make sense.

**Comment by flynnolivia** [comment]
Hi, I've got another use case in the Microsoft Edge Tools extension.

This would be helpful in order to comply with [WCAG 4.1.3](https://w3c.github.io/wcag21/understanding/status-messages.html), which is about status messages. Specifically, we need to announce when a list of inspectable targets has successfully finished refreshing. I think that @isidorn's idea about the silent notification with a flag for whether it'll be announced should be sufficient for this scenario.

Let me know if you have any suggestion for a work-around in the meantime, but hopefully this flag can be introduced.

**Comment by isidorn** [comment]
@flynnolivia I see. If I understand correctly your operation is a long running one, and once it is done you want to announce that?
Are you currently using the [`progress` API](https://github.com/microsoft/vscode/blob/main/src/vs/vscode.d.ts#L5619) if you are, the `progress` API should be fully accessible, so while the progress is ongoing (if it is longer than 1s) the screen reader while nicely read out the progress. Once the progress is done screen reader will announce that.
I am asking because this API sounds like a good fit for your use case.

**Comment by setaskin** [comment]
@isidorn  sorry for the late response.  Your suggestions makes sense. I have another scenario examples. We have multiple bugs opened via Accesibility test team. Most of them related to Narrator not announcing UI items. I will post the scenarios here . Here is one of them : 
-. Narrator not announcing placeholder string in quick pick item ( here narrator doesn't announce "what would you like to do with other collaborator?") . I can see same behavior in vs code quick pick items. 
![image](https://user-images.githubusercontent.com/37163639/113078174-6bcad700-9187-11eb-8d84-3858ae09d270.png)

**Comment by flynnolivia** [comment]
@isidorn I wanted to follow up on this since I tried out the `progress` API and had a few issues still. It seems that the announcement only comes through when using `location: ProgressLocation.Notification` and it is only announced with Narrator but not NVDA (even if I set the progress to last longer than 1 second).

**Comment by isidorn** [comment]
@setaskin can you also reproduce that with NVDA? I think NVDA should just nicely read out the placeholder text of the QuickPick

@flynnolivia thanks for trying it out. Would you use the `ProgressLocation.Notification` or what would be ideal for your extension? 
As for NVDA not announcing progress that sounds like a bug in NVDA. Maybe even this one https://github.com/nvaccess/nvda/issues/9606
I suggest to file a new issue to NVDA or to comment on that one

**Comment by flynnolivia** [comment]
@isidorn I think that showing the message in the status bar would be most appropriate. So, using `ProgressLocation.Window` or having a way to use `window.setStatusBarMessage()` where the message is announced and can be removed after a timeout. I'll check out that NVDA issue too, thanks!

**Comment by setaskin** [comment]
@isidorn . yes, NVDA is reading the placeholder text of the QuickPick, it is just the Narrator not reading it.

**Comment by isidorn** [comment]
@setaskin please file a bug against the narrator team then.
You can do that using internal tools. If you do not know how to file it, feel free to ping me on Teams and I can provide details.

**Comment by isidorn** [comment]
Same as my initial comment https://github.com/microsoft/vscode/issues/114718#issuecomment-771767306
Based on this I am closing this as out of scope since we do not plan to work on this

**Comment by fsteffi** [comment]
GitHubTags:#A11y_VSCode;#A11ySev3;#A11yTCS;#A11yWCAG2.1;#DesktopApp;#Visual Studio Code Client;#Narrator;#WCAG4.1.3;#Win32;#VisualStudioCode-Win32-Apr2021;#VoiceOver;
